packages/pwa/app/utils/url.test.js

```diff
@@ -109,88 +110,68 @@ describe('url builder test', () => {
         const url = categoryUrlBuilder({id: 'men'})
         expect(url).toEqual(`/category/men`)
     })
+})
 
-    test('homeUrlBuilder returns expect', () => {
-        getUrlConfig.mockImplementation(() => ({
-            locale: 'path'
-        }))
-        const url = homeUrlBuilder('/', 'fr-FR')
-        expect(url).toEqual(`/fr-FR/`)
-
-        const homeUrlDefaultLocale = homeUrlBuilder('/', DEFAULT_LOCALE)
-        expect(homeUrlDefaultLocale).toEqual(`/`)
-    })
-
+describe('getUrlWithLocale', () => {
     test('getUrlWithLocale returns expected for PLP', () => {
-        getUrlConfig.mockImplementation(() => ({
-            locale: 'path'
-        }))
-        const location = new URL('http://localhost:3000/it-IT/category/newarrivals-womens')
-
-        window.location = location
+        const location = new URL('http://localhost:3000/uk/it-IT/category/newarrivals-womens')
 
-        const relativeUrl = getUrlWithLocale('fr-FR')
-        expect(relativeUrl).toEqual(`/fr-FR/category/newarrivals-womens`)
+        const relativeUrl = getUrlWithLocale('fr-FR', {location, site: mockConfig.app.sites[0]})
```

Comment:
I am not sure I get your comment.

 In this test,
- Before, the url is `'http://localhost:3000/uk/it-IT/category/newarrivals-womens'`. 
-  After passing the url to `getUrlWithLocale` with fr-FR, the new url has been updated with `fr-FR` which is the expected behavior. 

We pass the site here to let it know how to sanitize the site from url as well as the locale so we can rebuild it with `buildPathWithUrlConfig` respectively.

Reply:
@alexvuong I was wondering why changing the `opts.site` value does not seem to do anything. If that's the case, can we remove it? So that the function signature has one less thing, like this: `getPathWithLocale('fr-FR', {location})`

The way we pass `opts` to the function `buildPathWithUrlConfig` will not do anything because it expects to see only `opts.disallowParams`.  
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/88cce7dca64cfcb3785ee0d64ab9baba584e683e/packages/pwa/app/utils/url.js#L155-L167

Reply:
@vmarta, I've removed the `opts.site`. The default site will be used in case the `siteRef` is not found in the url